acceptance/framework/consul/cli_cluster.go

```diff
@@ -0,0 +1,246 @@
+		// which could result in tests passing due to that token having privileges to read services
+		// (false positive).
+		"dns.enabled": "false",
+	}
+	valuesFromConfig, err := cfg.HelmValuesFromConfig()
+	require.NoError(t, err)
+
+	// Merge all helm values
+	mergeMaps(values, valuesFromConfig)
+	mergeMaps(values, helmValues)
+
+	logger := terratestLogger.New(logger.TestLogger{})
+
+	// Wait up to 15 min for K8s resources to be in a ready state. Increasing
+	// this from the default of 5 min could help with flakiness in environments
+	// like AKS where volumes take a long time to mount.
+	extraArgs := map[string][]string{
+		"install": {"--timeout", "15m"},
+	}
+
+	opts := &helm.Options{
+		SetValues:      values,
+		KubectlOptions: ctx.KubectlOptions(t),
+		Logger:         logger,
+		ExtraArgs:      extraArgs,
+	}
+	opts.KubectlOptions.Namespace = "consul"
+	return &CLICluster{
+		ctx:                ctx,
+		helmOptions:        opts,
+		namespace:          cfg.KubeNamespace,
+		values:             values,
+		releaseName:        releaseName,
+		kubernetesClient:   ctx.KubernetesClient(t),
+		kubeConfig:         cfg.Kubeconfig,
+		kubeContext:        cfg.KubeContext,
+		noCleanupOnFailure: cfg.NoCleanupOnFailure,
+		debugDirectory:     cfg.DebugDirectory,
+		logger:             logger,
+	}
+}
+
+func (h *CLICluster) Create(t *testing.T) {
+	t.Helper()
+
+	// Make sure we delete the cluster if we receive an interrupt signal and
+	// register cleanup so that we delete the cluster when test finishes.
+	helpers.Cleanup(t, h.noCleanupOnFailure, func() {
+		h.Destroy(t)
+	})
+
+	// Fail if there are any existing installations of the Helm chart.
+	//h.checkForPriorInstallations(t)
+	args := []string{"install"}
+	kubeconfig := h.kubeConfig
+	if kubeconfig != "" {
+		args = append(args, "-kubeconfig", kubeconfig)
+	}
+	kubecontext := h.kubeContext
+	if kubecontext != "" {
+		args = append(args, "-kubecontext", kubecontext)
+	}
+
+	for k, v := range h.values {
+		args = append(args, "-set", fmt.Sprintf("%s=%s", k, v))
+
+	}
+	args = append(args, "-auto-approve")
+	fmt.Println(args)
+	cmd := exec.Command("consul-k8s", args...)
+
+	require.NoError(t, cmd.Run())
+
+	helpers.WaitForAllPodsToBeReady(t, h.kubernetesClient, "consul", fmt.Sprintf("release=%s", h.releaseName))
+}
+
+func (h *CLICluster) Destroy(t *testing.T) {
+	t.Helper()
+
+	k8s.WritePodsDebugInfoIfFailed(t, h.helmOptions.KubectlOptions, h.debugDirectory, "release="+h.releaseName)
+
+	args := []string{"uninstall"}
+	kubeconfig := h.kubeConfig
+	if kubeconfig != "" {
+		args = append(args, "-kubeconfig", kubeconfig)
+	}
+	kubecontext := h.kubeContext
+	if kubecontext != "" {
+		args = append(args, "-kubecontext", kubecontext)
+	}
+
+	args = append(args, "-auto-approve")
+	args = append(args, "-wipe-data")
+	fmt.Println(args)
+	cmd := exec.Command("consul-k8s", args...)
+
+	require.NoError(t, cmd.Run())
+}
+
+func (h *CLICluster) Upgrade(t *testing.T, helmValues map[string]string) {
```

Comment:
From here on down in this file, the implementation is exactly the same as the HelmCluster implementation